puppet/modules/slave/manifests/postgresql.pp

```diff
@@ -1,22 +1,38 @@
 # @api private
 class slave::postgresql {
-  Postgresql_psql {
-    cwd => '/',
-  }
+  # only CentOS slaves are used to run unit tests
+  if $::osfamily == 'RedHat' {
+    if $::operatingsystemmajrelease == '7' {
 
-  include ::postgresql::client
-  include ::postgresql::server
+			exec { 'remove postgresql':
+				command => "/bin/yum -y remove postgresql-server postgresql-client postgresql",
+				before => Class['postgresql::globals'],
+			}
 
-  # Simple known user/pass that will allow Jenkins to create its
-  # own databases when required
-  postgresql::server::role { 'foreman':
-    password_hash => postgresql_password('foreman', 'foreman'),
-    superuser     => true,
-    login         => true,
-  }
+      if $::operatingsystem == 'CentOS' {
+      	package { 'centos-release-scl-rh':
+          ensure => 'present',
+          before => Class['postgresql::globals'],
+        }
+      } else {
+        exec { 'enable SCL repo from RHUI':
+      	  command => "/bin/yum-config-manager --enable rhel-server-rhui-rhscl-7-rpms",
+          before => Class['postgresql::globals'],
+        }
+      }
+
+      class { 'postgresql::globals':
+        version              => '12',
+        client_package_name  => 'rh-postgresql12-postgresql-syspaths',
+        server_package_name  => 'rh-postgresql12-postgresql-server-syspaths',
+        contrib_package_name => 'rh-postgresql12-postgresql-contrib-syspaths',
+        service_name         => 'postgresql',
+        datadir              => '/var/opt/rh/rh-postgresql12/lib/pgsql/data',
+        confdir              => '/var/opt/rh/rh-postgresql12/lib/pgsql/data',
+        bindir               => '/usr/bin',
+      }
+    }
 
-  # only CentOS slaves are used to run unit tests
```

Comment:
These tunings were made RH-only because Debian didn't like `checkpoint_segments` but now that's removed so they can be made generic and moved to the bottom.

Reply:
I don't see checkpoint segments anywhere.

Reply:
It was line 28. It didn't work on Debian since it was already PG >= 10 which dropped `checkpoint_segments`.